tests/graph/test_basic.py

```diff
@@ -302,6 +302,17 @@ def test_eval(self):
             pickle.loads(pickle.dumps(self.w)), "_fn_cache"
         ), "temporary functions must not be serialized"
 
+    def test_eval_with_strings(self):
+        assert self.w.eval({"x": 1.0, self.y: 2.0}) == 6.0
+        assert self.w.eval({self.z: 3}) == 6.0
+
+    def test_eval_errors_having_mulitple_variables_same_name(self):
```

Comment:
Should also add a test with zero variables matching since you have a specific error message for that case.

Reply:
Thank you for your review @ricardoV94. Will add the corresponding test case.